influxql group by behavior, measurement, and time columns (#5026)
influxdata/docs-v2: content/influxdb/cloud-serverless/query-data/influxql/basic-query.md

---
title: Perform a basic InfluxQL query
seotitle: Perform a basic InfluxQL query in InfluxDB Cloud
description: >
  A basic InfluxQL query that queries data from InfluxDB most commonly includes
  `SELECT`, `FROM`, and `WHERE` clauses.
menu:
  influxdb_cloud_serverless:
    name: Basic query
    parent: Query with InfluxQL
    identifier: query-influxql-basic
weight: 202
influxdb/cloud-serverless/tags: [query, influxql]
list_code_example: |
  ```sql
  SELECT temp, room FROM home WHERE time >= now() - 1d
  ```
---

InfluxQL (Influx Query Language) is an SQL-like query language used to interact
with InfluxDB and work with times series data.

A basic InfluxQL query that queries data from InfluxDB most commonly includes the
following clauses:

{{< req type="key" >}}

> - {{< req "\*">}} `SELECT`: Specify fields, tags, and calculations to output from a
>   measurement or use the wildcard alias (`*`) to select all fields and tags
>   from a measurement. It requires at least one [field key](/influxdb/cloud-serverless/reference/glossary/#field-key) or the wildcard alias (`*`). For more information, see [/influxdb/cloud-serverless/reference/influxql/select/#notable-select-statement-behaviors].

Reviewer: ```suggestion
- {{< req "\*">}} `SELECT`: Specify fields, tags, and calculations to return from a
  measurement or use the wildcard alias (`*`) to select all fields and tags
  from a measurement. It requires at least one
  [field key](/influxdb/cloud-serverless/reference/glossary/#field-key) or the wildcard alias (`*`).
  For more information, see [Notable SELECT statement behaviors](/influxdb/cloud-serverless/reference/influxql/select/#notable-select-statement-behaviors).
```

If a query uses a `GROUP BY` clause, the result set includes the following:

- Columns listed in the query's `SELECT` clause
- A `time` column that contains the timestamp for the record or the group
- An `iox::measurement` column that contains the record's measurement (table) name
- Columns listed in the query's `GROUP BY` clause; each row in the result set contains the values used for grouping

### GROUP BY result columns

If a query uses `GROUP BY` and the `WHERE` clause doesn't filter by time, then groups are based on the [default time range](/influxdb/cloud-serverless/reference/group-by/#default-time-range).

## Basic query examples

- [Query data within time boundaries](#query-data-within-time-boundaries)
- [Query data without time boundaries](#query-data-without-time-boundaries)
- [Query specific fields and tags](#query-specific-fields-and-tags)
- [Query fields based on tag values](#query-fields-based-on-tag-values)
- [Query points based on field values](#query-points-based-on-field-values)
- [Alias queried fields and tags](#alias-queried-fields-and-tags)

{{% note %}}
#### Sample data

> The following examples use the sample data written in the
> [Get started writing data guide](/influxdb/cloud-serverless/get-started/write/).
> To run the example queries and return results,
> [write the sample data](/influxdb/cloud-serverless/get-started/write/#write-line-protocol-to-influxdb)

Reviewer: ```suggestion
The following examples use the 
[Get started home sensor data](/influxdb/cloud-serverless/reference/sample-data/#get-started-home-sensor-data).
To run the example queries and return results,
[write the sample data](/influxdb/cloud-serverless/reference/sample-data/#write-the-home-sensor-data-to-influxdb)
```

to your {{% cloud-name %}} database before running the example queries.
{{% /note %}}

### Query data within time boundaries

- Use the `SELECT` clause to specify what tags and fields to return.
  Specify at least one field key.
  To return all tags and fields, use the wildcard alias (`*`).
- Specify the measurement to query in the `FROM` clause.
- Specify time boundaries in the `WHERE` clause.
  Include time-based predicates that compare the value of the `time` column to a timestamp.
  Use the `AND` logical operator to chain multiple predicates together.

{{% influxdb/custom-timestamps %}}
```sql
SELECT *
FROM home
WHERE
  time >= '2022-01-01T08:00:00Z'
  AND time <= '2022-01-01T12:00:00Z'
```
{{% /influxdb/custom-timestamps %}}

Query time boundaries can be relative or absolute.

{{< expand-wrapper >}}
{{% expand "Query with relative time boundaries" %}}

To query data from relative time boundaries, compare the value of the `time`
column to a timestamp calculated by subtracting an interval from a timestamp.
Use `now()` to return the timestamp for the current time (UTC).

##### Query all data from the last month

```sql

> SELECT * FROM home WHERE time >= now() - INTERVAL '1 month'

Reviewer: ```suggestion
SELECT * FROM home WHERE time >= now() - 30d
```

```

##### Query one day of data data from a week ago
```sql
SELECT *
FROM home
WHERE

>   time >= now() - INTERVAL '7 days'
>   AND time <= now() - INTERVAL '6 days'

Reviewer: ```suggestion
  time >= now() - 7d
  AND time <= now() - 6d
```

If a field or tag key includes special characters or spaces or is case-sensitive,
wrap the key in _double-quotes_.

```sql
SELECT time, room, temp, hum FROM home
```

### Query fields based on tag values

- In the `SELECT` clause, include fields you want to query and tags you want to base conditions on.
- In the `WHERE` clause, include predicates that compare the tag identifier to a string literal.
  Use [logical operators](/influxdb/cloud-serverless/reference/influxql/where/#logical-operators) to chain multiple predicates together and apply
  multiple conditions.

```sql
SELECT * FROM home WHERE room = 'Kitchen'
```

### Query points based on field values

- In the `SELECT` clause, include fields you want to query.
- In the `WHERE` clause, include predicates that compare the field identifier to a value or expression.
  Use [logical operators](/influxdb/cloud-serverless/reference/influxql/where/#logical-operators) (`AND`, `OR`) to chain multiple predicates together
  and apply multiple conditions.

```sql
SELECT co, time FROM home WHERE co >= 10 OR co <= -10
```

### Alias queried fields and tags

To alias or rename fields and tags that you query, use the `AS` clause.

> After the tag, field, or expression you want to alias, pass `AS` followed by the alias name as a string literal in double quotes (`"`)--for example:

Reviewer: ```suggestion
After the tag, field, or expression you want to alias, pass `AS` followed by the alias name as an identifier (wrap in double quotes (`"`) if the alias includes spaces or special characters)--for example:
```

```sql

> SELECT temp AS "temperature", hum AS "humidity" FROM home

Reviewer: ```suggestion
SELECT temp AS temperature, hum AS "humidity (%)" FROM home
```

```

{{% note %}}

> When aliasing columns in **InfluxQL**, use the `AS` clause and double-quoted alias names.

Reviewer: ```suggestion
When aliasing columns in **InfluxQL**, use the `AS` clause and an [identifier](/influxdb/cloud-serverless/reference/influxql/#identifiers).
```
